tap_square/sync.py

```diff
@@ -72,8 +81,21 @@ def sync(config, state, catalog):
                 )
                 singer.write_state(state)
 
+            elif tap_stream_id == 'customers':
```

Comment:
Why did this logic need to be inside the sync function as an if block? Why couldn't you push this down to the stream class sync logic? Would it require too much refactoring to avoid this?

Reply:
We wanted to put it in the class' sync function but putting only this one stream's syncing logic in there would make it an exception to the pattern the rest of the tap uses, so we opted to put it here to keep the tap consistent.

Reply:
I disagree with this decision. The whole point of the tap class's sync function is they are different classes, so overriding it there is the right way to do it. Other sync functions (for example, for incremental streams) are different. There's not just a single sync inside it...

And why did you merge without discussing this further?

Reply:
Please ignore the above comment after we zoom synced about this.